[release] v8.12.0 (#19698)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 8.12.0

_Sep 25, 2025_

We'd like to extend a big thank you to the 15 contributors who made this release possible. Here are some highlights ✨:

- ⌨️ Charts keyboard navigation

> - ⚡️ Charts: Add new `BatchScatter` component that can optionally replace `Scatter` for large datasets

Reviewer: We probably want to have a bigger highlight related to the charts/data grid integration.

@joserodolfofreitas @arminmeh 

Do we also want to add @alelthomas video here?

Reviewer 2: Yes, I think that the video would be great
And I think that https://github.com/mui/mui-x/pull/18021 should be in both grid premium and charts premium sections

Author: I've added `- 🤝 Grid-Charts integration` 🙌 
I'll wait for @alelthomas to confirm if the video is ready 👀

Reviewer 2: I don't see it in the latest diff.
In any case, I have approved the PR.
Everything except this point looks ok from the grid's perspective.
If the video is not ready for release, we can at least add a screenshot.

Author: Good news! The video is ready 🙌 

https://github.com/user-attachments/assets/28f1848e-dc85-4077-8756-a3e88afd4e54

- 🐞 Bugfixes
- 📚 Documentation improvements
- 🧰 Codemod requires Node >=20.19

`@mui/x-codemod` minimum supported Node version is `20.19`.

> This was only the case due to using the v18 `yargs` package; this only explicitly aligns with it. (#18979)

Reviewer 3: ```suggestion
This was only the case due to using the v18 `yargs` package; this now explicitly aligns with it. (#18979)
```

Special thanks go out to the community members for their valuable contributions:
@deade1e, @sai6855

The following are all team members who have contributed to this release:
@alexfauquette, @bernardobelchior, @flaviendelangle, @Janpot, @JCQuintas, @LukasTy, @michelengelen, @prakhargupta1, @rita-codes, @siriwatknp, @arminmeh, @romgrk

### Data Grid

#### `@mui/x-data-grid@8.12.0`

- [DataGrid] Fix flex column width diff calculation while resizing (#19667) @arminmeh

#### `@mui/x-data-grid-pro@8.12.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@8.12.0`.

#### `@mui/x-data-grid-premium@8.12.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@8.12.0`.

### Date and Time Pickers

#### `@mui/x-date-pickers@8.12.0`

- [pickers] Ensure reference value is not updated for invalid values (#19635) @michelengelen

> - [pickers] Fix `apiRef.current.isItemExpanded()` method (#19619) @flaviendelangle

Reviewer 4: ```suggestion
```

- [pickers] Preserve time format when using single column layout on `TimeRangePicker` (#19626) @sai6855

> - [pickers] Preserve time format when using single column layout on `DateTimePicker` and `DateTimeRangePicker` (#19608) @sai6855

Reviewer 3: ```suggestion
- [pickers] Preserve time format when using single column layout on  Date Time Picker and Date Time Range Picker (#19608) @sai6855
```

- [charts] Document how plugins can be used (#19343) @alexfauquette
- [charts] Export chart plugins per series type (#19337) @alexfauquette
- [charts] Export plugins (#19335) @alexfauquette
- [charts] Fix horizontal layout and toolbar (#19655) @alexfauquette
- [charts] Fix performance issue with JS animations (#19606) @bernardobelchior
- [charts] Fix piecewise scale causing wrong colors in axis with min/max (#19610) @bernardobelchior
- [charts] Fix zoom discard inconsistency (#19535) @bernardobelchior
- [charts] Introduce keyboard navigation (#19155) @alexfauquette
- [charts] Refactor `getAxisExtremum` (#19627) @bernardobelchior

#### `@mui/x-charts-pro@8.12.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@8.12.0`, plus:

- [charts-pro] Add `valueFormatter` to sankey (#19636) @JCQuintas
- [charts-pro] Allow `source/target` keywords in sankey link color (#19634) @JCQuintas
- [charts-pro] Allow exporting `SankeyChart` (#19659) @JCQuintas
- [charts-pro] Fix axis inversion when using axis `max` and `filterMode: 'discard'` (#19200) @bernardobelchior

### Tree View

#### `@mui/x-tree-view@8.12.0`

> - [tree view] Allow to pass null to the icon slots (#19569) @flaviendelangle

Reviewer 4: ```suggestion
- [tree view] Allow to pass null to the icon slots (#19569) @flaviendelangle
- [tree view] Fix `apiRef.current.isItemExpanded()` method (#19619) @flaviendelangle

```

- [docs] Add Charts example collection page (#18353) @prakhargupta1
- [docs] Add a Charts demo showcasing bar and scatter composition (#19605) @prakhargupta1
- [docs] Add composition Charts demo for legends and tooltip (#19602) @prakhargupta1
- [docs] Add recipe about server-side data export (#19617) @siriwatknp
- [docs] Clarify DataGrid layout requirements (#19413) @romgrk
- [docs] Fix `ExportServerSideData` demo layout shift (#19669) @siriwatknp
- [docs] Improve server-side `updateRow()` description (#19554) @deade1e
- [docs] Show how to customize drawing area background (#19682) @alexfauquette
- [docs] Add hook documentation pages (#19334) @Copilot

### Core

- [code-infra] Add copilot instructions specific to x repo (#19623) @JCQuintas
- [code-infra] Load `tsx` files in visual regression (#19595) @JCQuintas
- [code-infra] Remove renovate automerge (#19501) @Janpot
- [code-infra] Update `DEFAULT_TIMESTAMP` format to ISO 8601 (#19624) @Janpot
- [code-infra] Update `findLatestTaggedVersion` to filter tags based on major version (#19693) @michelengelen
- [docs-infra] Ensure `create-playground` script only runs if target file is absent (#19603) @michelengelen

> - [docs-infra] Add Prakharas a codeowner of the docs (#19679) @alexfauquette

Reviewer 3: ```suggestion
- [docs-infra] Add @prakhargupta1 as a codeowner of the docs (#19679) @alexfauquette
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 8.12.0

_Sep 25, 2025_

We'd like to extend a big thank you to the 15 contributors who made this release possible. Here are some highlights ✨:

- ⌨️ Charts keyboard navigation

> - ⚡️ Charts: Add new `BatchScatter` component that can optionally replace `Scatter` for large datasets

Reviewer 5: I'd rewrite it as such:

```suggestion
- ⚡️ Charts: Add new `renderer="svg-batch"` prop to Scatter charts that provides improved performance for large datasets
```

The `BatchScatter` component is an implementation detail. Users will enable this feature by setting the `renderer="svg-batch"` prop on `ScatterChart`

- [charts] Export plugins (#19335) @alexfauquette
- [charts] Fix horizontal layout and toolbar (#19655) @alexfauquette
- [charts] Fix performance issue with JS animations (#19606) @bernardobelchior
- [charts] Fix piecewise scale causing wrong colors in axis with min/max (#19610) @bernardobelchior
- [charts] Fix zoom discard inconsistency (#19535) @bernardobelchior
- [charts] Introduce keyboard navigation (#19155) @alexfauquette
- [charts] Refactor `getAxisExtremum` (#19627) @bernardobelchior

#### `@mui/x-charts-pro@8.12.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@8.12.0`, plus:

- [charts-pro] Add `valueFormatter` to sankey (#19636) @JCQuintas
- [charts-pro] Allow `source/target` keywords in sankey link color (#19634) @JCQuintas
- [charts-pro] Allow exporting `SankeyChart` (#19659) @JCQuintas
- [charts-pro] Fix axis inversion when using axis `max` and `filterMode: 'discard'` (#19200) @bernardobelchior

#### `@mui/x-charts-premium@8.12.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

> Same changes as in `@mui/x-charts-pro@8.12.0`.

Reviewer: ```suggestion
Same changes as in `@mui/x-charts-pro@8.12.0`, plus:

- [charts-premium] Grid-Charts integration (#18021) @arminmeh
```

- 🐞 Bugfixes
- 📚 Documentation improvements
- 🧰 Codemod requires Node >=20.19

`@mui/x-codemod` minimum supported Node version is `20.19`.
This was only the case due to using the v18 `yargs` package; this now explicitly aligns with it. (#18979)

Special thanks go out to the community members for their valuable contributions:
@deade1e, @sai6855, @thomas-mcdonald

The following are all team members who have contributed to this release:
@alexfauquette, @bernardobelchior, @flaviendelangle, @Janpot, @JCQuintas, @LukasTy, @michelengelen, @prakhargupta1, @rita-codes, @siriwatknp, @arminmeh, @romgrk

### Data Grid

#### `@mui/x-data-grid@8.12.0`

- [DataGrid] Fix flex column width diff calculation while resizing (#19667) @arminmeh

#### `@mui/x-data-grid-pro@8.12.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@8.12.0`.

#### `@mui/x-data-grid-premium@8.12.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@8.12.0`, plus:

> - [charts-premium] Grid-Charts integration (#18021) @arminmeh

Reviewer 2: ```suggestion
- [DataGridPremium] Grid-Charts integration (#18021) @arminmeh
- [DataGridPremium] Fix sorting and filtering of the tree group columns with aggregation (#19607) @arminmeh
- [DataGridPremium] Disable aggregation on the grouping column by default (#19692) @arminmeh
- [DataGridPremium] Do not rely on the group separation constant to retrieve the column name for the charts panel (#19677) @arminmeh
- [DataGridPremium] Fix stale aggregation state (#19690) @arminmeh
```

#### `@mui/x-date-pickers-pro@8.12.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@8.12.0`.

### Charts

#### `@mui/x-charts@8.12.0`

- [charts] Add batch renderer for scatter chart (#19075) @bernardobelchior
- [charts] Add renderer suffix to performance benchmarks (#19620) @bernardobelchior
- [charts] Document how plugins can be used (#19343) @alexfauquette
- [charts] Export chart plugins per series type (#19337) @alexfauquette
- [charts] Export plugins (#19335) @alexfauquette
- [charts] Fix horizontal layout and toolbar (#19655) @alexfauquette
- [charts] Fix performance issue with JS animations (#19606) @bernardobelchior
- [charts] Fix piecewise scale causing wrong colors in axis with min/max (#19610) @bernardobelchior
- [charts] Fix zoom discard inconsistency (#19535) @bernardobelchior
- [charts] Introduce keyboard navigation (#19155) @alexfauquette
- [charts] Refactor `getAxisExtremum` (#19627) @bernardobelchior
- [charts] Remove unused code path from `getAxisScale` (#19673) @bernardobelchior

> - [charts] Make new hideLegend prop on ChartWrapper optional (#19694) @thomas-mcdonald

Reviewer 5: ```suggestion
- [charts] Make new hideLegend prop on ChartWrapper optional (#19694) @thomas-mcdonald
- [charts] Fix chart crash in test environment (#19711) @JCQuintas 
```

### Data Grid

#### `@mui/x-data-grid@8.12.0`

- [DataGrid] Fix flex column width diff calculation while resizing (#19667) @arminmeh

#### `@mui/x-data-grid-pro@8.12.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@8.12.0`.

#### `@mui/x-data-grid-premium@8.12.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@8.12.0`, plus:

- [DataGridPremium] Grid-Charts integration (#18021) @arminmeh
- [DataGridPremium] Fix sorting and filtering of the tree group columns with aggregation (#19607) @arminmeh
- [DataGridPremium] Disable aggregation on the grouping column by default (#19692) @arminmeh
- [DataGridPremium] Do not rely on the group separation constant to retrieve the column name for the charts panel (#19677) @arminmeh
- [DataGridPremium] Fix stale aggregation state (#19690) @arminmeh
- [DataGridPremium] Fix pivot column being hidden on autosizing (#19699) @cherniavskii

### Date and Time Pickers

#### `@mui/x-date-pickers@8.12.0`

> - [pickers] Ensure reference value is not updated for invalid values (#19635) @michelengelen

Reviewer 4: ```suggestion
- [pickers] Ensure reference value is not updated for invalid values (#19635) @michelengelen
- [pickers] Fix `slotProps.textField.slotProps.htmlInput` resolution (#19713) @LukasTy
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 8.12.0

_Sep 25, 2025_

We'd like to extend a big thank you to the 15 contributors who made this release possible. Here are some highlights ✨:

> - 🤝 Grid-Charts integration
>   <img width="875" height="818" alt="Screenshot 2025-09-25 at 15 03 40" src="https://github.com/user-attachments/assets/98f55ab0-7d23-4b81-99fc-19a5fd825d01" />
>   👉 [🎥 Watch the full video](https://github.com/user-attachments/assets/28f1848e-dc85-4077-8756-a3e88afd4e54)

Reviewer: ```suggestion
- 🤝 Grid-Charts integration

  <img width="875" height="600" alt="Screenshot 2025-09-25 at 15 03 40" src="https://github.com/user-attachments/assets/98f55ab0-7d23-4b81-99fc-19a5fd825d01" />

  👉 [🎥 Watch the full video](https://github.com/user-attachments/assets/28f1848e-dc85-4077-8756-a3e88afd4e54)
```

Make image a bit smaller, divide lines with a blank line should make it look better

Author: now it looks niiiice
<img width="875" height="818" alt="Screenshot 2025-09-25 at 16 36 54" src="https://github.com/user-attachments/assets/62768451-d4b6-439c-a3f2-4c82fe54faf2" />

Reviewer 2: The only thing that needs a change is
`alt="Screenshot 2025-09-25 at 15 03 40"`

Reviewer 6: Nitpicky but can you add a wider screenshot so it looks more like a thumbnail? 
<img width="990" height="738" alt="Grid x Charts integration" src="https://github.com/user-attachments/assets/bdcecd3a-ad6c-4fbf-ada0-3e3add0f14e5" />

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 8.12.0

_Sep 25, 2025_

We'd like to extend a big thank you to the 15 contributors who made this release possible. Here are some highlights ✨:

- 🤝 Grid-Charts integration

>   <img width="500" alt="Screenshot 2025-09-25 at 15 03 40" src="https://github.com/user-attachments/assets/98f55ab0-7d23-4b81-99fc-19a5fd825d01" />

Reviewer: ```suggestion
  <img width="500" alt="Text reads "Auto-generate Charts from your Grid.", then the DataGrid charts integration in use, showing its configurator screen and a pie chart." src="https://github.com/user-attachments/assets/98f55ab0-7d23-4b81-99fc-19a5fd825d01" />
```
